### DISCUSSION TITLE
[@types/react-router-dom] type Location is not generic in useLocation

### DISCUSSION THREAD

**Discussion by byungmin12** [DefinitelyTyped/DefinitelyTyped#60159] [Issues with a @types package] [answered]
currently i'm using @types/react-dom(v18.0.3) and react-router-dom on my project. 

but i have some problem
```js
//way1
  const { state } = useLocation<{test:string}>();
```
it showed me err message
```
Expected 0 type arguments, but got 1
```

```js
//way2
  const { state } = useLocation() as Location<{test : string}>
```
```
Type 'Location' is not generic.
```

```
// imperfect solution :<
const { state } = useLocation()
//const state: unknown
```
this message is caused by type Location 
type Location is not generic. so i had no choice. i only had type unknown

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment, marked as answer]
`^5.1` definitely supports the type argument: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d7d679c285cf5e817ba79fc319d7aa4bc3547bbe/types/react-router/index.d.ts#L191

Please include where the types for `react-router` are coming from by running `npm ls @types/react-router` (when using NPM) or `yarn why @types/react-router` (when using Yarn).

**Reply by byungmin12** [reply]
thanks for your comment . 
well... can you explain what i do ? 

i installed @types/react-router. and i have  "@types/react-router": "^5.1.18" 

unfortunately it was still here  :< 

cuz useLocation has type Location.

**Reply by eps1lon** [reply]
I didn't mean to imply you should install them. Just to run one of these commands depending on which package manager you're using.

**Comment by 1pete** [comment]
Which version of `react-router` and `react-router-dom` are you using?

If it's v6+, it has builtin type definitions, you have to uninstall `@types/react-router` and `@types/react-router-dom` from your `package.json`. If the problem still persists, report it to https://github.com/remix-run/react-router/issues
